packages/block-editor/src/components/block-lock/style.scss

```diff
@@ -0,0 +1,56 @@
+.block-editor-block-lock-modal {
+	z-index: z-index(".block-editor-block-lock-modal");
+
+	.components-modal__frame {
+		@include break-small() {
+			max-width: $break-mobile;
+		}
+	}
+}
+
+.block-editor-block-lock-modal__checklist {
+	margin: 0;
+}
+
+.block-editor-block-lock-modal__options-title {
```

Comment:
@ciampo @mirka a few thing to check here in the context of not having to write CSS for basic UIs!

Reply:
Thank you for the ping!

Ideally, when encountering such limitations, styles overrides should be our last choice. Instead, we could:

- understand together the limitations encountered
- assess if the requirements make sense, given the current design language / components system
- understand if there's a way to achieve the requirements that doesn't involve CSS overrides
- assess if we should make any changes to the native components to enable any new functionality

@Mamaduka , what limitations did you exactly encounter when using `@wordpress/components` that forced you to override styles with custom CSS?

Reply:
Design/styles are based on the Block Manager component. The main limitation I can think of is the checklist and how labels icons are aligned. I don't believe `CheckboxControl` supports this out of the box.

P.S. I might not be most up to date on new best practices when building components. Happy push update if you can share documentation or examples.

Reply:
The generic rules that come to mind are:

- limit styles overrides as much as possible
- if in need of doing so, avoid relying on the components internal hardcoded classnames or their internal DOM structure

The more we override styles this way, the more coupled our code gets with the components' internal implementation details (hardcoded classnames and DOM structured are not part of a component's public API).

With respect to this specific PR, a few things that come off the top of my head:

- have you considered using [`ItemGroup`](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-itemgroup--default) instead of vanilla `ul` and `li` ?
- have you considered proposing changes to the `CheckboxControl` component directly?

Reply:
Thanks for the feedback, @ciampo.

> have you considered using [ItemGroup](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-itemgroup--default) instead of vanilla ul and li ?

I've not, but I will give it a try.

> have you considered proposing changes to the CheckboxControl component directly?

I'm not sure if these specific changes will be helpful.

Reply:
@ciampo

I tried using `ItemGroup`/`Item` instead of `ul` and `li` instead of vanilla. It was easy to replace them. But I still have to use style overrides for `CheckboxControl`, so I don't think change is worth it at the moment.

Maybe we should try refactoring the locking modal and BlockManager components as a follow-up.

Reply:
Sure thing, let's iterate on this in a follow-up PR!

Just a note to future selves: The styles overrides currently seems to be there in order to:

1. Tweak look&feel of `CheckboxControl`:
    - Apply typography styles to the label
    - Apply layout/vertical alignment overriders between the icon and the label
2. Style a vertical list of items (spacing, borders...)


Potential solutions to look into:
- for the typography styles, look into the `Text` or `Heading` components;
- for list layout, look into the `ItemGroup` / `Item` components (or into the `Flex` components for a lower-level component)
- take a look at the `CheckboxControl` component and see if we can make any amends there (or even add a way to customise its layout/alignment without overriding styles)